#!/bin/bash

#PBS -l walltime=23:59:00
#PBS -l nodes=1:ppn=12
#PBS -l mem=12gb
#PBS -r n
#PBS -j oe
#PBS -A ujp-495-aa

cd $SCRATCH/sophie_breton/genome/abyss/blast


module load blast+
module load BWA
module load SAMtools

#blastn the 
#Venustaconcha_ellipsiformis_mito.blastout

#blastn the transcriptome ?
#blastn -num_threads 12 -query /RQexec/renaut/sophie_breton/genome/sequences/rnaseq/Trinity.fasta -db Venustaconcha_ellipsiformis_mito.fasta -outfmt 6 -out Venustaconcha_ellipsiformis_transcripto_vs_mito.blastout
#RESULTS: most the mito is there, split in about 12 contigs.


#####OPTION 1
#align all the reads against the reference transcriptome?
#bwa index ../results/dir_k41_B24G_H4_kc3/assembly_k41_B24G_H4_kc3-10.fa
#bwa index Venustaconcha_ellipsiformis_mito.fasta

#align reads unto the mito genome and pipe it to samtools view to only get mapped reads
bwa mem -t 12 Venustaconcha_ellipsiformis_haplotype_female.fasta ../../sequences/pairends/err_corrected_bfc_HI.0502.006.Index_10.H3412_9_R1_paired.fastq.gz ../../sequences/pairends/err_corrected_bfc_HI.0502.006.Index_10.H3412_9_R2_paired.fastq.gz | samtools view -F4 >female_mapped6.sam

bwa mem -t 12 Venustaconcha_ellipsiformis_haplotype_female.fasta ../../sequences/pairends/err_corrected_bfc_HI.0502.007.Index_10.H3412_9_R1_paired.fastq.gz ../../sequences/pairends/err_corrected_bfc_HI.0502.007.Index_10.H3412_9_R2_paired.fastq.gz | samtools view -F4 >female_mapped7.sam

bwa mem -t 12 Venustaconcha_ellipsiformis_haplotype_female.fasta ../../sequences/pairends/err_corrected_bfc_HI.0502.008.Index_10.H3412_9_R1_paired.fastq.gz ../../sequences/pairends/err_corrected_bfc_HI.0502.008.Index_10.H3412_9_R2_paired.fastq.gz | samtools view -F4 >female_mapped8.sam

cat female_mapped6.sam female_mapped7.sam female_mapped8.sam >testassembly_mapped.sam

#get mapped reads
#samtools view -F4 Venustaconcha_ellipsiformis_mito.sam  >testassembly_mapped.sam

#get header
#head -2 Venustaconcha_ellipsiformis_mito.sam >head.sam

#cat header + mapped reads
cat head.sam testassembly_mapped.sam >testassembly_mapped2.sam
#rm testassembly_mapped.sam head.sam

#sam to bam
samtools view -Sb testassembly_mapped2.sam >testassembly_mapped2.bam

#sort by read name
samtools sort -n testassembly_mapped2.bam  -o testassembly_mapped2_sorted.bam

#bamtofastq (all mapped reads)
/RQusagers/renaut/bedtools2/bin/bedtools bamtofastq -i testassembly_mapped2_sorted.bam -fq mito.fq

#then align reads unto the mito using bwa, or make a denovo assembly using "ABYSS"
~/abyss/bin/ABYSS -k 42 mito.fq -o mito.out

bwa mem Venustaconcha_ellipsiformis_haplotype_female.fasta mito.fq >mito.sam
samtools view -Sb mito.sam >mito.bam
samtools sort mito.bam >mito_sort.bam
samtools index mito_sort.bam 
samtools tview mito_sort.bam Venustaconcha_ellipsiformis_haplotype_female.fasta


#RESULTS: there are about 5000 reads in mito.fq, they represent several (4-5) section of the mito genome, but most of the mito genome is absent


####OPTION 2
#map the contigs from the final assembly onto the mito genome
#bwa mem -t 12 Venustaconcha_ellipsiformis_mito.fasta ../results/dir_k41_B24G_H4_kc3/assembly_k41_B24G_H4_kc3-10.fa >Venustaconcha_ellipsiformis_mito.sam

#RESULTS: few reads map

###OPTION 3 (preferred)
#blast assembled genome onto the mito

#RESULTS: a single section around the NADH.
#blastn -num_threads 12 -query ../results/dir_k41_B24G_H4_kc3/assembly_k41_B24G_H4_kc3-10.fa  -db Venustaconcha_ellipsiformis_mito.fasta -outfmt 6 Venustaconcha_ellipsiformis_mito.blastout
